src/Utilities/TypeTraits.hpp

```diff
@@ -906,6 +906,39 @@ using is_callable_t = typename is_callable<T, Args...>::type;
   using is_##METHOD_NAME##_callable_t =                                        \
       is_##METHOD_NAME##_callable_r_t<AnyReturnType##METHOD_NAME, T, Args...>;
 
+/*!
+ * \ingroup TypeTraitsGroup
+ * \brief Generate a type trait to check if a class has a `static constexpr`
+ * variable, optionally also checking its type.
+ *
+ * \example
+ * \snippet Utilities/Test_TypeTraits.cpp CREATE_HAS_EXAMPLE
+ *
+ * \see `CREATE_IS_CALLABLE`
+ */
+#define CREATE_HAS(CONSTEXPR_NAME)                                           \
+  struct NoSuchType##CONSTEXPR_NAME {};                                      \
+                                                                             \
+  template <typename CheckingType,                                           \
+            typename VariableType = NoSuchType##CONSTEXPR_NAME,              \
+            typename = cpp17::void_t<>>                                      \
+  struct has_##CONSTEXPR_NAME : std::false_type {};                          \
+                                                                             \
+  template <typename CheckingType, typename VariableType>                    \
+  struct has_##CONSTEXPR_NAME<CheckingType, VariableType,                    \
+                              cpp17::void_t<std::remove_const_t<decltype(    \
+                                  CheckingType::CONSTEXPR_NAME)>>>           \
+      : cpp17::disjunction<                                                  \
```

Comment:
`cpp17::bool_constant<cpp17::is_same_v<....> or ...>`

Reply:
for my education, what's the difference?

Reply:
Modern compiler implementations are able to avoid several type constructions when using the variable versions (`_v`), which disjunction doesn't do. Furthermore, the way disjunction has to be implemented makes it rather horrid with regards to compile time, unfortunately.